src/hooks/auth0-provider.tsx

```diff
@@ -16,10 +25,10 @@ const initialState = {
 /**
  * @ignore
  */
-const getIdTokenProfileClaims = idToken => {
-  const payload = jwt_decode(idToken);
+const getIdTokenProfileClaims = (idToken: string): User => {
+  const payload: {[key: string]: any} = jwtDecode<JwtPayload>(idToken);
```

Comment:
Is there a specific reason we exit from the JwtPayload type to move to an object that can hold anything? We do lose type-safety on all the JwtPayload defined properties.

Reply:
This is because `JwtPayload` currently doesn't support custom claims that we will use to get the profile information like `username`, `phone_number` etc

Reply:
You can extend that:
```ts
type CustomJwtPayload = JwtPayload & { {[key: string]: any} }

const payload: {[key: string]: any} = jwtDecode<CustomJwtPayload>(idToken);

```